Userland/Libraries/LibWeb/HTML/ImageBitmap.h

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (c) 2024, Lucas Chollet <lucas.chollet@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#pragma once
+
+#include <AK/OwnPtr.h>
+#include <LibGfx/Bitmap.h>
+#include <LibWeb/Bindings/PlatformObject.h>
+#include <LibWeb/Bindings/Serializable.h>
+#include <LibWeb/Bindings/Transferable.h>
+#include <LibWeb/Forward.h>
+#include <LibWeb/HTML/Canvas/CanvasDrawImage.h>
+
+namespace Web::HTML {
+
+using ImageBitmapSource = Variant<CanvasImageSource, JS::Handle<FileAPI::Blob>, JS::Handle<ImageData>>;
+
+// https://html.spec.whatwg.org/multipage/imagebitmap-and-animations.html#imagebitmapoptions
+struct ImageBitmapOptions {
+    // FIXME: Implement these fields
+};
+
+class ImageBitmap final : public Bindings::PlatformObject
+    , public Web::Bindings::Serializable
+    , public Web::Bindings::Transferable {
+    WEB_PLATFORM_OBJECT(ImageBitmap, Bindings::PlatformObject);
+    JS_DECLARE_ALLOCATOR(ImageBitmap);
+
+public:
+    static JS::NonnullGCPtr<ImageBitmap> create(JS::Realm&);
+    virtual ~ImageBitmap() override = default;
+
+    // ^Web::Bindings::Serializable
+    virtual StringView interface_name() const override { return "ImageBitmap"sv; }
+    virtual WebIDL::ExceptionOr<void> serialization_steps(HTML::SerializationRecord&, bool for_storage, HTML::SerializationMemory&) override;
+    virtual WebIDL::ExceptionOr<void> deserialization_steps(ReadonlySpan<u32> const&, size_t& position, HTML::DeserializationMemory&) override;
+
+    // ^Web::Bindings::Transferable
+    virtual WebIDL::ExceptionOr<void> transfer_steps(HTML::TransferDataHolder&) override;
+    virtual WebIDL::ExceptionOr<void> transfer_receiving_steps(HTML::TransferDataHolder&) override;
+    virtual HTML::TransferType primary_interface() const override;
+
+    unsigned width() const;
```

Comment:
minor, but `WebIDL::UnsignedLong` for this, and the member (which currently mismatch each other anyway :P)